### PR TITLE
Do not override the values of non-formatting options when a profile is set

### LINKED ISSUE #1991: The testsuite fails with OCaml 4.14
```
$ dune runtest -p ocamlformat
# File "test/passing/tests/js_source.ml.ref", line 1, characters 0-0:
#          git (internal) (exit 1)
# (cd _build/default && /usr/bin/git --no-pager diff --no-index --color=always -u test/passing/tests/js_source.ml.ref test/passing/js_source.ml.stdout)
# diff --git a/test/passing/tests/js_source.ml.ref b/test/passing/js_source.ml.stdout
# index d81f988..c68d3a8 100644
# --- a/test/passing/tests/js_source.ml.ref
# +++ b/test/passing/js_source.ml.stdout
# @@ -542,9 +542,9 @@ let _ = Obj.extension_constructor 7 (* Invald_arg *)
#  
#  let _ =
#    Obj.extension_constructor
# -    (object
# -       method m = 3
# -    end)
# +    object
# +      method m = 3
# +    end
#  ;;
#  
#  (* Invald_arg *)
# @@ -1398,27 +1398,27 @@ and (_, _) ty_case =
#  
#  let ty_abc : (([ `A of int | `B of string | `C ] as 'a), 'e) ty =
#    Sum
# -    (object
# -       method proj =
# -         function
# -         | `A n -> "A", Some (Tdyn (Int, n))
# -         | `B s -> "B", Some (Tdyn (String, s))
# -         | `C -> "C", None
# -
# -       method cases =
# -         [ "A", TCarg (Thd, Int)
# -         ; "B", TCarg (Ttl Thd, String)
# -         ; "C", TCnoarg (Ttl (Ttl Thd))
# -         ]
# -
# -       method inj
# -           : type c.  (int -> string -> noarg -> unit, c) ty_sel * c
# -                     -> [ `A of int | `B of string | `C ] =
# -         function
# -         | Thd, v -> `A v
# -         | Ttl Thd, v -> `B v
# -         | Ttl (Ttl Thd), Noarg -> `C
# -    end)
# +    object
# +      method proj =
# +        function
# +        | `A n -> "A", Some (Tdyn (Int, n))
# +        | `B s -> "B", Some (Tdyn (String, s))
# +        | `C -> "C", None
# +
# +      method cases =
# +        [ "A", TCarg (Thd, Int)
# +        ; "B", TCarg (Ttl Thd, String)
# +        ; "C", TCnoarg (Ttl (Ttl Thd))
# +        ]
# +
# +      method inj
# +          : type c.  (int -> string -> noarg -> unit, c) ty_sel * c
# +                    -> [ `A of int | `B of string | `C ] =
# +        function
# +        | Thd, v -> `A v
# +        | Ttl Thd, v -> `B v
# +        | Ttl (Ttl Thd), Noarg -> `C
# +    end
#  ;;
#  
#  type 'a vlist =
# @@ -1431,19 +1431,19 @@ let ty_list : type a e. (a, e) ty -> (a vlist, e) ty =
#    let tcons = Pair (Pop t, Var) in
#    Rec
#      (Sum
# -       (object
# -          method proj =
# -            function
# -            | `Nil -> "Nil", None
# -            | `Cons p -> "Cons", Some (Tdyn (tcons, p))
# -
# -          method cases = [ "Nil", TCnoarg Thd; "Cons", TCarg (Ttl Thd, tcons) ]
# -
# -          method inj : type c. (noarg -> a * a vlist -> unit, c) ty_sel * c -> a vlist =
# -            function
# -            | Thd, Noarg -> `Nil
# -            | Ttl Thd, v -> `Cons v
# -       end))
# +       object
# +         method proj =
# +           function
# +           | `Nil -> "Nil", None
# +           | `Cons p -> "Cons", Some (Tdyn (tcons, p))
# +
# +         method cases = [ "Nil", TCnoarg Thd; "Cons", TCarg (Ttl Thd, tcons) ]
# +
# +         method inj : type c. (noarg -> a * a vlist -> unit, c) ty_sel * c -> a vlist =
# +           function
# +           | Thd, Noarg -> `Nil
# +           | Ttl Thd, v -> `Cons v
# +       end)
#  ;;
#  
#  (*
```
cc @gpetiot

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #1991
Replacing #1994

### CODE REVIEW

File: bin/ocamlformat-rpc/main.ml [hunk 1]
```diff
@@ -127,7 +126,18 @@ let rec rpc_main = function
       | `Handled v ->
           Init.output stdout (`Version vstr) ;
           Out_channel.flush stdout ;
-          rpc_main (Version_defined (v, Conf.default_profile))
+          rpc_main
+            (Version_defined
+               ( v
+               , { fmt_opts= Conf.default_profile
+                 ; opr_opts=
```
Comment (Julow) [inline bin/ocamlformat-rpc/main.ml]: We need a `default_opts` somewhere.

File: lib/Translation_unit.ml [hunk 2]
```diff
@@ -291,31 +292,35 @@ let format (type a b) (fg : a Extended_ast.t) (std_fg : b Std_ast.t)
...
     in
-    if opts.debug then
+    if conf.opr_opts.debug then
       format ~box_debug:true |> fst
       |> dump_formatted ~suffix:".boxes"
       |> (ignore : string option -> unit) ;
     let fmted, cmts_t = format ~box_debug:false in
-    let conf = if opts.debug then conf else {conf with quiet= true} in
+    let conf =
+      if conf.opr_opts.debug then conf
```
Comment (Julow) [inline lib/Translation_unit.ml:317]: This `if` is not necessary, the else branch covers all the cases.

**Review by Julow — APPROVED** [review]
Looks good :)

But all this to be able to specify `comment_check`, `ocaml_version`, etc.. in attributes ? I think the only non-formatting option we still want in attributes is `disable` and I don't think it needs to be in `Conf.t` either.

**Comment by gpetiot** [comment]
> But all this to be able to specify `comment_check`, `ocaml_version`, etc.. in attributes ? I think the only non-formatting option we still want in attributes is `disable` and I don't think it needs to be in `Conf.t` either.

Told you this would be a big change! :p

From the options in `opr_opts` only `disable` can be set in attributes because operational options have `allow_inline=false` by default. Having these options in `Conf.t` allows to set them in `.ocamlformat` files, this was already the case for the ones in the old `Conf.t` but not for the ones that were built with `mk ...`, we could only set them on cmdline.

File: bin/ocamlformat-rpc/main.ml (hunk 1 above)
Comment (gpetiot) [reply bin/ocamlformat-rpc/main.ml]: done

**Comment by kit-ty-kate** [comment]
OCaml-CI confirms this fixes the tests on 4.14

File: lib/Translation_unit.ml (hunk 2 above)
Comment (Julow) [reply lib/Translation_unit.ml:317]: It is not! Ignore this.

**Pull request merged**